src/alire/alire-config.adb

```diff
@@ -33,4 +103,297 @@ package body Alire.Config is
+         Trace.Error ("Using default: '" & Image (Default) & "'");
+         return Default;
+
+      else
+         return TOML_As_Return_Type (Val.Value);
+      end if;
+   end Get_With_Default_Gen;
+
+   ---------
+   -- Get --
+   ---------
+
+   function Get (Key     : Config_Key;
+                 Default : Boolean)
+                 return Boolean
+   is
+      function Get_With_Default_Bool is new Get_With_Default_Gen
+        (Boolean, TOML_Boolean, TOML.As_Boolean, Boolean'Image);
+
+   begin
+      return Get_With_Default_Bool (Key, Default);
+   end Get;
+
+   ---------
+   -- Get --
+   ---------
+
+   function Get (Key     : Config_Key;
+                 Default : String)
+                 return String
+   is
+      function Id (Str : String) return String
+      is (Str);
+
+      function Get_With_Default_Str is new Get_With_Default_Gen
+        (String, TOML_String, TOML.As_String, Id);
+
+   begin
+      return Get_With_Default_Str (Key, Default);
+   end Get;
+
+   ---------
+   -- Get --
+   ---------
+
+   function Get (Key     : Config_Key;
+                 Default : TOML.Any_Integer)
+                 return TOML.Any_Integer
+   is
+      function Get_With_Default_Int is new Get_With_Default_Gen
+        (TOML.Any_Integer, TOML_Integer, TOML.As_Integer, Any_Integer'Image);
+
+   begin
+      return Get_With_Default_Int (Key, Default);
+   end Get;
+
+   ---------
+   -- Get --
+   ---------
+
+   function Get (Key     : Config_Key;
+                 Default : TOML.Any_Float)
+                 return TOML.Any_Float
+   is
+      function Get_With_Default_Int is new Get_With_Default_Gen
+        (TOML.Any_Float, TOML_Float, TOML.As_Float, Image);
+   begin
+      return Get_With_Default_Int (Key, Default);
+   end Get;
+
+   ----------
+   -- List --
+   ----------
+
+   function List (Filter      : String := ".*";
+                  Show_Origin : Boolean := False)
+                  return String
+   is
+      use GNAT.Regexp;
+
+      Re     : constant Regexp := Compile (Filter, Glob => True);
+
+      Result : Unbounded_String;
+   begin
+      for C in Config_Map.Iterate loop
+         declare
+            Val : constant Config_Value := Config_Map (C);
+            Key : constant String := To_String (Config_Maps.Key (C));
+         begin
+            if Match (Key, Re) then
+               if Show_Origin then
+                  Append (Result, Val.Source & " (" & Val.Lvl'Img & "): ");
+               end if;
+
+               Append (Result, Key & "=");
+               Append (Result, Image (Val.Value));
+               Append (Result, ASCII.LF);
+            end if;
+         end;
+      end loop;
+      return To_String (Result);
+   end List;
+
+   --------------
+   -- Filepath --
+   --------------
+
+   function Filepath (Lvl : Level) return Absolute_Path is
+   begin
+      case Lvl is
+         when Global =>
+            return Alire.Platform.Default_Config_Folder / "config.toml";
```

Comment:
I suspect this should be `Config.Path`, or the ability to use different global configurations (`-c` switch) will be broken